### PR TITLE
[MI-2549] Add plugin API to get assignment groups

### PR DESCRIPTION
**Pull request by ayusht2810**

### CODE REVIEW

File: server/constants/routes.go [hunk 1]
```diff
@@ -22,15 +22,17 @@ const (
...
-	PathGetUserFromServiceNow           = "/api/now/table/sys_user"
-	PathGetIncidentFieldsFromServiceNow = "api/" + ServiceNowForMattermostNotificationsAppID + "/getincidentfields"
+	PathActivateSubscriptions             = "api/now/table/" + ServiceNowForMattermostNotificationsAppID + "_servicenow_for_mattermost_notifications_auth"
+	PathSubscriptionCRUD                  = "api/now/table/" + ServiceNowForMattermostNotificationsAppID + "_servicenow_for_mattermost_subscriptions"
+	PathGetRecordsFromServiceNow          = "api/now/table/{tableName}"
+	PathGetStatesFromServiceNow           = "api/" + ServiceNowForMattermostNotificationsAppID + "/getstates/{record_type}"
+	PathGetCatalogItemsFromServiceNow     = "api/sn_sc/servicecatalog/items"
+	PathGetUserFromServiceNow             = "/api/now/table/sys_user"
+	PathGetIncidentFieldsFromServiceNow   = "api/" + ServiceNowForMattermostNotificationsAppID + "/getincidentfields"
+	PathGetAssignmentGroupsFromServiceNow = "api/now/table/sys_user_group"
```
Comment (raghavaggarwal2308) [inline server/constants/routes.go:35]: Should we replace "_" with "-"?

**Review by raghavaggarwal2308 — APPROVED** [review]

File: server/constants/routes.go (hunk 1 above)
Comment (ayusht2810) [reply server/constants/routes.go:35]: it is the table name

File: server/plugin/client.go [hunk 2]
```diff
@@ -310,3 +311,21 @@ func (c *client) GetIncidentFieldsFromServiceNow() ([]*serializer.ServiceNowInci
...
 	return fields.Result, statusCode, nil
 }
+
+func (c *client) SearchAssignmentGroupsInServiceNow(searchTerm, limit, offset string) ([]*serializer.ServiceNowAssignmentGroup, int, error) {
+	query := fmt.Sprintf("%s LIKE%s", constants.FieldName, searchTerm)
+	queryParams := url.Values{
+		constants.SysQueryParam:       {query},
+		constants.SysQueryParamLimit:  {limit},
+		constants.SysQueryParamOffset: {offset},
+		constants.SysQueryParamFields: {fmt.Sprintf("%s,%s,%s", constants.FieldSysID, constants.FieldName, constants.FieldDescription)},
```
Comment (manojmalik20) [inline server/plugin/client.go:321]: What is the need for "description"?

**Review by manojmalik20 — APPROVED** [review]

File: server/plugin/client.go (hunk 2 above)
Comment (ayusht2810) [reply server/plugin/client.go:321]: it is fixed in next PR
https://github.com/Brightscout/mattermost-plugin-servicenow/blob/da0904ad8109dcba1cdeefea25b190dd585b8573/server/plugin/client.go#L321

**Pull request merged**